gobblin-runtime/src/main/java/gobblin/runtime/JobState.java

```diff
@@ -317,6 +344,12 @@ public void addTaskStates(Collection<TaskState> taskStates) {
     }
   }
 
+  public void addSkippedTaskStates(Collection<TaskState> taskStates) {
+    for (TaskState taskState : taskStates) {
+      this.skippedTaskStates.put(taskState.getTaskId(), taskState);
```

Comment:
It might be simpler to inline this code in the initial iteration itself. Else, make it a method that works on a single element rather than the entire Collection so that it can be called from the first iteration.